uefi-test-runner/src/proto/console/gop.rs

```diff
@@ -48,29 +48,39 @@ fn fill_color(gop: &mut GraphicsOutput) {
 fn draw_fb(gop: &mut GraphicsOutput) {
     let mi = gop.current_mode_info();
     let stride = mi.stride();
-    // BUG: we should check we have enough space to draw.
-    // let (width, height) = mi.resolution();
+    let (width, height) = mi.resolution();
 
     let fb = unsafe { gop.frame_buffer() };
 
-    let mut set_pixel = |(row, column), (r, g, b)| {
-        let index = (row * stride) + column;
-
-        // BUG: we assume the pixel format is 32-bit BGR, as it often is on x86.
-        // For RGB the red / blue channels will be inverted.
```

Comment:
Fixed. Simplified the functionality of set_pixel along the way to reduce code duplication between RGB and BGR.